cpp/src/text/edit_distance.cu

```diff
@@ -245,49 +237,51 @@ std::unique_ptr<cudf::column> edit_distance_matrix(cudf::strings_column_view con
         d_strings.is_null(row) ? cudf::string_view{} : d_strings.element<cudf::string_view>(row);
       cudf::string_view const d_str2 =
         d_strings.is_null(col) ? cudf::string_view{} : d_strings.element<cudf::string_view>(col);
-      if (d_str1.empty() || d_str2.empty()) return;
-      // the temp size needed is 3 int16s per character of the shorter string
-      d_offsets[idx - ((row + 1) * (row + 2)) / 2] = std::min(d_str1.length(), d_str2.length()) * 3;
+      if (d_str1.empty() || d_str2.empty()) { return; }
+      // the temp size needed is 2 integers per character of the shorter string
+      d_offsets[idx - ((row + 1) * (row + 2)) / 2] =
+        (std::min(d_str1.length(), d_str2.length()) + 1) * 2;
     });
 
   // get the total size for the compute buffer
-  size_t compute_size =
-    thrust::reduce(rmm::exec_policy(stream), offsets.begin(), offsets.end(), size_t{0});
+  int64_t compute_size =
+    thrust::reduce(rmm::exec_policy(stream), offsets.begin(), offsets.end(), int64_t{0});
   // convert sizes to offsets in-place
   thrust::exclusive_scan(rmm::exec_policy(stream), offsets.begin(), offsets.end(), offsets.begin());
   // create the compute buffer
-  rmm::device_uvector<int16_t> compute_buffer(compute_size, stream);
+  rmm::device_uvector<cudf::size_type> compute_buffer(compute_size, stream);
   auto d_buffer = compute_buffer.data();
 
   // compute the edit distance into the output column
-  auto results   = cudf::make_fixed_width_column(cudf::data_type{cudf::type_id::INT32},
+  auto results = cudf::make_fixed_width_column(cudf::data_type{cudf::type_to_id<cudf::size_type>()},
                                                strings_count * strings_count,
                                                rmm::device_buffer{0, stream, mr},
                                                0,
                                                stream,
                                                mr);
-  auto d_results = results->mutable_view().data<int32_t>();
+  auto d_results = results->mutable_view().data<cudf::size_type>();
   thrust::for_each_n(
     rmm::exec_policy(stream),
     thrust::make_counting_iterator<cudf::size_type>(0),
     strings_count * strings_count,
```

Comment:
are we ignoring any overflow issues? do we need to use different type here & at other places to avoid such overflows?

Reply:
Thanks. looks good. nothing missed.